boards/rpi-pico/include/periph_conf.h

```diff
@@ -28,6 +28,16 @@
 extern "C" {
 #endif
 
+/**
+ * @brief   Silences the warning when an unsigned value is compared to 0
+ *
+ * This can be deleted when I2C is properly implemented.
```

Comment:
I'm still dreaming of a proper low-level I2C driver interface so multiple backends are possible without hacks like this.

I think it can even be done without changing the API and maybe even a fast path if only a single backend is selected.